src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionProviderFactory.cs

```diff
@@ -25,7 +25,7 @@ internal static string GetNamespaceFromFQN(string fullyQualifiedName)
             return namespaceName;
         }
 
-        internal static RazorCodeAction CreateAddUsingCodeAction(string fullyQualifiedName, DocumentUri uri)
+        internal static RazorCodeAction CreateAddUsingCodeAction(string fullyQualifiedName, DocumentUri uri, bool enforceCodeActionInvokedInComponent)
```

Comment:
Is there harm in just allowing the code actions outside of components? I think for many of the component light bulbs they'll end up no-oping

Reply:
> allowing the code actions outside of components

The *razor* code actions? If so, sure, just did this to preserve existing behavior.

Reply:
I see. If they don't actually cause any harm in non-component files then I wouldn't sweat it.

Reply:
On second thought, it would add the namespace for a component, to the top of a non-component file, this doesn't really seem like ideal behavior? Maybe best to leave as is? Sorry for the confusion.

Reply:
> On second thought, it would add the namespace for a component, to the top of a non-component file, this doesn't really seem like ideal behavior? Maybe best to leave as is? Sorry for the confusion.

Hmm add a namespace from what light bulb?

Reply:
🤯 you're right. Removed the check.